### DISCUSSION TITLE
Is it possible to record console session with clink?

### DISCUSSION THREAD

**Discussion by CKylinMC** [chrisant996/clink#463] [General]
Just like PowerSession(or asciinema), is it possible to record CMD console session with clink in any way?

**Comment by chrisant996** [comment]
Please explain what is meant by "recode".  I don't want to have to go do a bunch of research to figure out what is being asked.  Please share all of the relevant information here.  Thanks!

**Reply by CKylinMC** [reply]
Yes, I mean “record”, sorry for the misspelling!

**Reply by chrisant996** [reply]
@CKylinMC can you clarify what you want to record?

Which of these do you mean?

- [ ] the prompts that are shown
- [ ] the command lines that are typed as input
- [ ] the output from the commands that are executed
    - [ ] just the text that shows up on screen
    - [ ] also the raw output that includes ANSI escape codes

I assume you want all of them, and that's not possible.

**Reply by CKylinMC** [reply]
I don't know what step it can do, even with modern terminal program like Windows Terminal. If you could do all these features, that will be awesome, but I also DO know it's hard to do so. For me, prompts, and my inputs, and basic text output is enough for me. The problem is “ANSI escape codes”, colors are not necessary for my case, but if some cli programs using flush line and rewrite something in previous lines, that should be matter. 
I'm using cmd in Windows Terminal only, will it be easier to implement these things in modern terminal?

**Reply by CKylinMC** [reply]
I want to record: what I input, what it reacted, what I input after that.

**Reply by chrisant996** [reply]
> I'm using cmd in Windows Terminal only, will it be easier to implement these things in modern terminal?

If you mean "is it easier to implement in Windows Terminal itself?" -- then yes.

If you mean "is it easier for Clink to implement it if Windows Terminal provides the window for cmd.exe?" -- then no.

You want to capture all output to a console window, from all processes anywhere that attach to it or write output.  Clink just runs inside one instance of cmd.exe running in the console window -- but there can be many other processes all using the same console.  What you're asking for would need to be done in the terminal program itself.  Try making the suggestion in the Windows Terminal repo instead.  👍

**Reply by CKylinMC** [reply]
Well, thank you for your patience in explaining! Clink is a super cool utility and helped me a lot in my daily, thanks for your awesome works on it!

**Comment by chrisant996** [comment]
Do you mean "record" instead of "recode"?

Clink does not capture the console session and cannot capture the output from other programs in the console session.

If that's not what you mean, then please describe in more detail what you are asking for.

Thanks!

**Reply by CKylinMC** [reply]
Yes, I mean “record” CMD sessions. I'm finding a tool like "PowerSession" but for cmd, to record my console sessions and could replay later. I know it's nearly impossible, but after I use clink for a few months, I think “clink seems extended cmd features, maybe there is a way to do that thing?”, after I search for it, I found nothing. I don't want to miss something, so I open this discussion here...
